soc/nordic/common/pm_s2ram.S

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ /**
+ * @file
+ * @brief Nordic suspend-to-RAM code (S2RAM)
+ */
+
+#include <zephyr/toolchain.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/arch/common/pm_s2ram.h>
+
+
+GTEXT(pm_s2ram_mark_set)
+SECTION_FUNC(TEXT, pm_s2ram_mark_set)
+	/*
+	 * Restore the PC and continue
+	 */
+	mov r15, r1
+
+GTEXT(pm_s2ram_mark_check_and_clear)
+SECTION_FUNC(TEXT, pm_s2ram_mark_check_and_clear)
+	/*
+	 * Set return value to 0
+	 */
+	mov	r0, #0
+
+	/*
+	 * Load check RESETREAS register
+	 */
+	ldr	r2, =CONFIG_NRF_RESETINFO_PERIPH_ADDRESS
```

Comment:
Ideally we should avoid putting MDK stuff in kconfig. What about using inline assembly in c file? Then you can provide data from MDK.

Reply:
> Ideally we should avoid putting MDK stuff in kconfig. What about using inline assembly in c file? Then you can provide data from MDK.

Can we be sure that such a function won't push anything onto the stack?